acceptance/tests/agent/last_run_summary_report.rb

```diff
@@ -0,0 +1,108 @@
+
+          # Known issue on Windows: 'C:\ProgramData\PuppetLabs\puppet' permissions are inherited
+          # by its subfolders and it does not have any permissions for 'Everyone' (see 'PuppetAppDir'
+          # in 'puppet-agent/resources/windows/wix/appdatafiles.wxs')
+          # Below line should be added when solution is found:
+          # assert_match(/Everyone:.*\(RX\)/, result.stdout)
+        end
+      end
+    end
+
+    step "Create the 'last_run_summary.yaml' report file by applying catalog" do
+      on(agent, puppet('agent -t')) do |result|
+        assert_match('Applied catalog', result.stdout)
+      end
+    end
+
+    step "Check if the 'last_run_summary.yaml' report file created has '0644' permissions" do
+      unless agent['platform'] =~ /windows/
+        on(agent, "ls -al #{publicdir}") do |result|
+          assert_match(/rw-r--r--.+last_run_summary\.yaml$/, result.stdout)
+        end
+      else
+        on(agent, "icacls #{File.join(publicdir, 'last_run_summary.yaml')}") do |result|
+          # Linux 'Owner' premissions class equivalent
+          assert_match('Administrator:(R,W', result.stdout)
+          # Linux 'Group' permissions class equivalent
+          assert_match('None:(R)', result.stdout)
+          # Linux 'Public' permissions class equivalent
+          assert_match('Everyone:(R)', result.stdout)
+        end
+      end
+    end
+
+    step "Check that '#{statedir}' exists and has no 'last_run_summary.yaml' file" do
+      on(agent, "ls #{statedir}",:acceptable_exit_codes => [0, 2]) do |result|
+        assert_no_match(/No such file or directory/, result.stderr)
+        assert_no_match(/last_run_summary.yaml/, result.stdout)
+      end
+    end
+    
+    step "Check that 'publicdir' can be reconfigured" do
+      on(agent, puppet("config set publicdir #{custom_publicdir}"))
+      on(agent, puppet('config print publicdir')) do |result|
+        assert_match(custom_publicdir, result.stdout)
+      end
+    end
+
+    step "Create a new 'last_run_summary.yaml' report file by applying catalog" do
+      on(agent, puppet('agent -t')) do |result|
+        assert_match('Applied catalog', result.stdout)
+      end
+    end
+
+    step "Check if the 'last_run_summary.yaml' report file was created in the new location and still has '0644' permissions" do
+      unless agent['platform'] =~ /windows/
+        on(agent, "ls -al #{custom_publicdir}") do |result|
+          assert_match(/rw-r--r--.+last_run_summary\.yaml$/, result.stdout)
+        end
+      else
+        on(agent, "icacls #{File.join(custom_publicdir, 'last_run_summary.yaml')}") do |result|
+          # Linux 'Owner' premissions class equivalent
+          assert_match('Administrator:(R,W', result.stdout)
+          # Linux 'Group' permissions class equivalent
+          assert_match('None:(R)', result.stdout)
+          # Linux 'Public' permissions class equivalent
+          assert_match('Everyone:(R)', result.stdout)
+        end
```

Comment:
I think switching to if/else and change the code blocks would make this easier to read